CI 这边 **Windows checks 挂了**（macOS / Linux / Android cargo check 都过了），和 PR 描述里「Windows IMMNotificationClient 分支需 CI 验证」对上了。

**失败点（`device_watch.rs`）**
```
error[E0432]: unresolved import `windows::Win32::Media::Audio::IMMNotificationClient_Impl`
error[E0432]: unresolved import `windows::Win32::UI::Shell::PropertiesSystem`
error[E0433]: cannot find `windows_core` in the crate root   (#[windows::core::implement(...)] 两处)
```

**修复方向**
1. **`Cargo.toml` windows features**：当前只有 `"Win32_UI_Shell"`，建议补上 `"Win32_UI_Shell_PropertiesSystem"`（`PROPERTYKEY` / `OnPropertyValueChanged` 签名依赖它）。
2. **`windows-core` 依赖**：`#[windows::core::implement(IMMNotificationClient)]` 需要 crate 根能解析 `windows_core`，Windows target 下加 `windows-core = "0.58"`（与现有 `windows = "0.58"` 对齐）通常就能过。
3. **`IMMNotificationClient_Impl` import**：windows-rs 0.58 里一般不需要手动 `use ...IMMNotificationClient_Impl`；`#[implement(IMMNotificationClient)]` 会生成 `DeviceNotificationClient_Impl`，直接 `impl IMMNotificationClient_Impl for DeviceNotificationClient_Impl` 即可——若仍报错，对照同版本 windows-rs 示例调整 import 路径。

macOS CoreAudio 路径和 60s 兜底设计看起来 OK；Windows 编译修好后我再看 CI。Thanks!

--- a/openless-all/app/src-tauri/src/device_watch.rs
+++ b/openless-all/app/src-tauri/src/device_watch.rs
@@ -3,11 +3,11 @@
 //! 目的：替代「每 10s 轮询 `list_input_devices()`」这条空闲唤醒。改用各平台原生的设备
 //! 变更通知，空闲时零唤醒，设备插拔/默认设备切换时由 OS 回调实时触发刷新。
 //!
-//! 严格按平台分流：
+//! 平台分流：
 //! - macOS：CoreAudio `AudioObjectAddPropertyListener` 监听
 //!   `kAudioHardwarePropertyDevices`，专用线程跑 `CFRunLoop` 常驻。
-//! - Windows：`IMMNotificationClient` 经 `IMMDeviceEnumerator` 注册端点通知回调。
-//! - Linux：无原生路径，返回 `false`，由 `lib.rs` 的慢速兜底轮询负责。
+//! - Windows / Linux：暂返回 `false`，由 `lib.rs` 的 60s 慢速兜底轮询负责。
+//!   （Windows 原生 `IMMNotificationClient` 通知留作后续，需 Windows 开发机验证。）
 //!
 //! 三平台共用契约：`spawn_native_watcher(app, on_change)`。`on_change` 是 `lib.rs` 提供
 //! 的去抖闭包（内部复用 `microphone_device_signature()`，真正变化才刷新+emit）。回调里
@@ -27,17 +27,9 @@ where
     macos::spawn(on_change)
 }
 
-/// Windows：经 MMDevice 端点通知注册。见模块内说明。
-#[cfg(target_os = "windows")]
-pub(crate) fn spawn_native_watcher<F>(_app: AppHandle, on_change: F) -> bool
-where
-    F: Fn() + Send + Sync + 'static,
-{
-    windows_impl::spawn(on_change)
-}
-
-/// Linux：无原生路径，纯靠 `lib.rs` 的慢速兜底轮询。
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
+/// 非 macOS（Windows / Linux）：暂无本地验证过的原生路径，返回 `false`，纯靠 `lib.rs`
+/// 的 60s 慢速兜底轮询。Windows 原生 `IMMNotificationClient` 留作后续（需 Windows 开发机验证）。
+#[cfg(not(target_os = "macos"))]
 pub(crate) fn spawn_native_watcher<F>(_app: AppHandle, _on_change: F) -> bool
 where
     F: Fn() + Send + Sync + 'static,
@@ -172,167 +164,3 @@ mod macos {
     }
 }
 
-// ===================================================================================
-// Windows — IMMNotificationClient via IMMDeviceEnumerator
-// ===================================================================================
-//
-// 注意：开发机是 macOS，本分支无法本地编译验证，正确性靠 CI + 真机。严格 cfg-gate，
-// 写法保守稳妥。关键约束：
-//   1. enumerator 必须在监听期间一直存活，否则回调停止——故由常驻线程持有，退出时才
-//      `UnregisterEndpointNotificationCallback` + drop。
-//   2. COM 在本线程 `CoInitializeEx(MULTITHREADED)`，退出时 `CoUninitialize`。
-//   3. 只关心 capture（输入）设备的增删/状态/默认变化；render 设备变化忽略。回调里只调
-//      `on_change`（其内部去抖会过滤掉无关变化），故即使偶尔多触发也只是多一次零副作用
-//      的签名比较。
-#[cfg(target_os = "windows")]
-mod windows_impl {
-    use std::sync::atomic::Ordering;
-    use std::time::Duration;
-
-    use windows::core::Result as WinResult;
-    use windows::core::PCWSTR;
-    use windows::Win32::Media::Audio::{
-        eCapture, EDataFlow, ERole, IMMDeviceEnumerator, IMMNotificationClient,
-        IMMNotificationClient_Impl, MMDeviceEnumerator, DEVICE_STATE,
-    };
-    use windows::Win32::System::Com::{
-        CoCreateInstance, CoInitializeEx, CoUninitialize, CLSCTX_ALL, COINIT_MULTITHREADED,
-    };
-    use windows::Win32::UI::Shell::PropertiesSystem::PROPERTYKEY;
-
-    use super::super::TRAY_MICROPHONE_WATCHER_STOPPING;
-
-    /// IMMNotificationClient 实现：相关回调都转交 `on_change`（其内部去抖）。
-    /// 拥有 `on_change` 的所有权（'static Box），故 client 自包含、不借用外部生命周期。
-    #[windows::core::implement(IMMNotificationClient)]
-    struct DeviceNotificationClient {
-        on_change: Box<dyn Fn() + Send + Sync>,
-    }
-
-    impl DeviceNotificationClient {
-        fn notify(&self) {
-            (self.on_change)();
-        }
-    }
-
-    // IMMNotificationClient 的全部方法都必须实现（COM 接口契约）。
-    #[allow(non_snake_case)]
-    impl IMMNotificationClient_Impl for DeviceNotificationClient_Impl {
-        fn OnDeviceStateChanged(
-            &self,
-            _pwstrDeviceId: &PCWSTR,
-            _dwNewState: DEVICE_STATE,
-        ) -> WinResult<()> {
-            // 设备启用/禁用/拔出。无法在此廉价判断 data-flow，统一交给签名去抖过滤。
-            self.notify();
-            Ok(())
-        }
-
-        fn OnDeviceAdded(&self, _pwstrDeviceId: &PCWSTR) -> WinResult<()> {
-            self.notify();
-            Ok(())
-        }
-
-        fn OnDeviceRemoved(&self, _pwstrDeviceId: &PCWSTR) -> WinResult<()> {
-            self.notify();
-            Ok(())
-        }
-
-        fn OnDefaultDeviceChanged(
-            &self,
-            flow: EDataFlow,
-            _role: ERole,
-            _pwstrDefaultDeviceId: &PCWSTR,
-        ) -> WinResult<()> {
-            // 默认设备变化：只关心 capture（输入）。render 变化忽略，减少无谓刷新。
-            if flow == eCapture {
-                self.notify();
-            }
-            Ok(())
-        }
-
-        fn OnPropertyValueChanged(
-            &self,
-            _pwstrDeviceId: &PCWSTR,
-            _key: &PROPERTYKEY,
-        ) -> WinResult<()> {
-            // 属性变化噪声大（音量等），不触发刷新。
-            Ok(())
-        }
-    }
-
-    /// 在专用线程注册 MMDevice 端点通知并常驻直到退出 flag 置位。
-    /// 成功返回 `true`；COM/注册失败返回 `false`。
-    pub(super) fn spawn<F>(on_change: F) -> bool
-    where
-        F: Fn() + Send + Sync + 'static,
-    {
-        // 闭包提前装箱：client 直接拥有它，全程不借用外部生命周期，无裸指针 hack。
-        let on_change: Box<dyn Fn() + Send + Sync> = Box::new(on_change);
-        let (tx, rx) = std::sync::mpsc::channel::<bool>();
-        let spawn_result = std::thread::Builder::new()
-            .name("openless-mic-mmdevice".into())
-            .spawn(move || {
-                // SAFETY: 在本专用线程初始化 COM（MTA）。失败则不注册，直接报失败退出。
-                let hr = unsafe { CoInitializeEx(None, COINIT_MULTITHREADED) };
-                if hr.is_err() {
-                    log::warn!("[device_watch] CoInitializeEx failed: {hr:?}");
-                    let _ = tx.send(false);
-                    return;
-                }
-
-                let registered = register_and_run(on_change);
-                let _ = tx.send(registered);
-
-                // SAFETY: 与上面的 CoInitializeEx 配对。register_and_run 返回时所有 COM 对象
-                // 已 drop（enumerator/client 在其作用域内），此处可安全反初始化 COM。
-                unsafe { CoUninitialize() };
-            });
-
-        if let Err(err) = spawn_result {
-            log::warn!("[device_watch] spawn MMDevice watcher thread failed: {err}");
-            return false;
-        }
-        rx.recv().unwrap_or(false)
-    }
-
-    /// 创建 enumerator、注册回调、常驻轮转检查退出 flag、退出前注销。注册成功返回 `true`。
-    ///
-    /// 关键：本函数返回前 `enumerator` 与 `client` 一直在作用域内存活 → 满足「监听期间
-    /// enumerator 必须存活，否则回调停」。`client` 拥有 `on_change` 的所有权，自包含。
-    fn register_and_run(on_change: Box<dyn Fn() + Send + Sync>) -> bool {
-        // SAFETY: 标准 MMDevice enumerator 实例化。CLSCTX_ALL 允许进程内/外服务器。
-        let enumerator: IMMDeviceEnumerator =
-            match unsafe { CoCreateInstance(&MMDeviceEnumerator, None, CLSCTX_ALL) } {
-                Ok(e) => e,
-                Err(err) => {
-                    log::warn!(
-                        "[device_watch] CoCreateInstance(MMDeviceEnumerator) failed: {err:?}"
-                    );
-                    return false;
-                }
-            };
-
-        let client: IMMNotificationClient = DeviceNotificationClient { on_change }.into();
-
-        // SAFETY: enumerator 有效；client 是合法 IMMNotificationClient。注册后系统持有 client
-        // 的引用计数，回调将从 MMDevice 线程进入。client 拥有 on_change，生命周期自洽。
-        if let Err(err) = unsafe { enumerator.RegisterEndpointNotificationCallback(&client) } {
-            log::warn!("[device_watch] RegisterEndpointNotificationCallback failed: {err:?}");
-            return false;
-        }
-
-        // 常驻：MMDevice 回调由系统线程驱动，本线程只需保持 enumerator/client 存活并周期
-        // 检查退出 flag。1s 一片，退出最多 1s 内生效。
-        while !TRAY_MICROPHONE_WATCHER_STOPPING.load(Ordering::Relaxed) {
-            std::thread::sleep(Duration::from_secs(1));
-        }
-
-        // SAFETY: 注销同一 client。enumerator 仍有效。注销后系统不再回调。
-        if let Err(err) = unsafe { enumerator.UnregisterEndpointNotificationCallback(&client) } {
-            log::warn!("[device_watch] UnregisterEndpointNotificationCallback failed: {err:?}");
-        }
-        // enumerator 与 client 在此 drop（COM 引用计数归还，on_change 随 client 释放）。
-        true
-    }
-}

